source/common/upstream/outlier_detection_impl.cc

```diff
@@ -348,8 +351,9 @@ void DetectorImpl::checkHostForUneject(HostSharedPtr host, DetectorHostMonitorIm
       runtime_.snapshot().getInteger(BaseEjectionTimeMsRuntime, config_.baseEjectionTimeMs()));
   const std::chrono::milliseconds max_eject_time = std::chrono::milliseconds(
       runtime_.snapshot().getInteger(MaxEjectionTimeMsRuntime, config_.maxEjectionTimeMs()));
+  const std::chrono::milliseconds jitter = monitor->getJitter();
   ASSERT(monitor->numEjections() > 0);
-  if ((min(base_eject_time * monitor->ejectTimeBackoff(), max_eject_time)) <=
+  if ((min(base_eject_time * monitor->ejectTimeBackoff() + jitter, max_eject_time + jitter)) <=
```

Comment:
nit:
```suggestion
  if ((min(base_eject_time * monitor->ejectTimeBackoff(), max_eject_time) + jitter) <=
```

Reply:
I agree with all the comments. I added a commit with fixes